Tests/Functional/bootstrap/FeatureContext.php

```diff
@@ -0,0 +1,241 @@
+     */
+    public function jeConvertisEnCsvLeTableauContenuDans($filename, $prefix = null)
+    {
+        $filepath = realpath($this->requests_path . "/" . $filename);
+
+        $array_to_convert = file_get_contents($filepath);
+        $array_to_convert = json_decode($array_to_convert, true);
+
+        if ($array_to_convert === null) {
+            throw new Exception("json_decode error");
+        }
+
+        $array_to_convert = array_map(
+            function($line) use ($prefix) {
+                return CsvUtils::getTokenFromArray($line, $prefix);
+            },
+            $array_to_convert
+        );
+
+        try {
+            $this->result = CsvUtils::arrayToCsv($array_to_convert, $this->csv_lines);
+        } catch (\Exception $exception) {
+            $this->error = $exception;
+        }
+
+    }
+
+    /**
+     * @When /^j'envoie un mail a "([^"]*)" avec "([^"]*)" avec le titre "([^"]*)" et le template contenu dans le fichier "([^"]*)" et les tokens contenus dans "([^"]*)"$/
+     */
+    public function jEnvoieUnMail($to_email, $from_email, $title, $template_filename, $tokens_filename)
+    {
+        $template_filepath = realpath($this->requests_path . "/" . $template_filename);
+        $template          = file_get_contents($template_filepath);
+        $tokens_filepath   = realpath($this->requests_path . "/" . $tokens_filename);
+        $tokens_content    = file_get_contents($tokens_filepath);
+        $tokens            = json_decode($tokens_content, true);
+
+        try {
+            NotifyUtils::sendMail(self::$silex_app, $title, $template, $from_email, $to_email, $tokens);
+        } catch (\Exception $exception) {
+            $this->error = $exception;
+        }
+    }
+
+    /**
+     * @When /^je lance une impression avec le template "([^"]*)" et les données contenues dans "([^"]*)"(?: dans la queue "([^"]*)"?)?$/
+     */
+    public function jeLanceUneImpression($template_filename, $tokens_filename, $queue_name = null)
+    {
+        $template_filepath = realpath($this->requests_path . "/" . $template_filename);
+        $template          = file_get_contents($template_filepath);
+        $tokens_filepath   = realpath($this->requests_path . "/" . $tokens_filename);
+        $tokens_content    = file_get_contents($tokens_filepath);
+        $tokens            = json_decode($tokens_content, true);
+
+        try {
+            NotifyUtils::sendPrint(self::$silex_app, $template_filename, $template, $queue_name, $tokens);
+        } catch (\Exception $exception) {
+            $this->error = $exception;
+        }
+    }
+
+    /**
+     * @Then /^il doit y avoir un message dans la file "([^"]*)" avec le corps contenu dans "([^"]*)"$/
+     */
+    public function ilDoitYavoirUnMessageDansLaFileAvecLeCorpsContenuDans($queue = null, $body = null)
+    {
+        if ($body !== null) {
+            if (!file_exists($this->results_path . $body)) {
```

Comment:
vu que tu l'utilises plusieurs fois, tu pourrais en faire une variable de `$this->results_path . $body`